public/agg_types/lib/custom_date_handler.js

```diff
@@ -0,0 +1,26 @@
+import moment from 'moment';
+import dateMath from '@elastic/datemath';
+
+/**
+ * Rounds up `customComparing.to` to the end of the day if needed.
+ * Returns the current time (`moment()`) if the field is missing.
+ *
+ * @param { from, to } customComparing
+ */
+export function handleCustomDate(customComparing) {
+  // If customComparing is missing,
+  //  returns the object as if the fields are empty.
+  if (!customComparing) return { from: moment(), to: moment() };
+
+  // Checks if `customComparing.to` is using day format.
+  //  If so, rounds it to the end of the day
+  const isEndDateUsingTime = customComparing.to && customComparing.to.includes(':');
+  const momentEndDate = moment(customComparing.to);
+  const isEndDateInDayFormat = !isEndDateUsingTime && momentEndDate.isSame(momentEndDate.startOf('day'));
```

Comment:
great!

Though it does't work with `2018-05-45 00`. It will be rounded to the end of the day when it shouldn't.